lib/screens/home/home.dart

```diff
@@ -1,122 +1,40 @@
 import 'package:tavern/widgets/search_bar.dart';
+import 'package:tavern/widgets/tavern_logo.dart';
 
 class Home extends StatefulWidget {
   @override
   _HomeState createState() => _HomeState();
 }
 
 class _HomeState extends State<Home> with TickerProviderStateMixin {
-  final _htmlParsingClient = PubHtmlParsingClient();
-
-  Page firstPage;
-
-  /// Defaults to 1 for the first page
-  int currentPage = 1;
-
-  List<FullPackage> packagesFromPage = [];
+  HomeBloc homeBloc;
 
   final _scaffoldKey = GlobalKey<ScaffoldState>();
 
   String titleFilter = "Top";
 
-  int _currentSelection = 2;
-
-  ScrollController _scrollController;
-
   SortType sortType = SortType.overAllScore;
   FilterType filterType = FilterType.all;
-
-  void _handleFilterSelection(String selection, List<Package> packages) {
-    switch (selection) {
-      case 'Overall Score':
-        setState(() {
-          titleFilter = 'Top';
-          sortType = SortType.overAllScore;
-        });
-        break;
-      case 'Recently Updated':
-        setState(() {
-          titleFilter = 'Updated';
-          sortType = SortType.recentlyUpdated;
-        });
-        break;
-      case 'Newest Package':
-        setState(() {
-          titleFilter = 'New';
-          sortType = SortType.newestPackage;
-        });
-        break;
-      case 'Popularity':
-        setState(() {
-          titleFilter = 'Popular';
-          sortType = SortType.popularity;
-        });
-        break;
-      default:
-        break;
-    }
-  }
-
-  void loadPreferences() async {
-    SharedPreferences prefs = await SharedPreferences.getInstance();
-    String sort = prefs.get('FeedSortSelection');
-    setState(() {
-      switch (sort) {
-        case 'OverallScore':
-          sortType = SortType.overAllScore;
-          titleFilter = "Top";
-          break;
-        case 'RecentlyUpdated':
-          sortType = SortType.recentlyUpdated;
-          titleFilter = 'Updated';
-          break;
-        case 'NewestPackage':
-          sortType = SortType.newestPackage;
-          titleFilter = 'New';
-          break;
-        case 'Popularity':
-          sortType = SortType.popularity;
-          titleFilter = 'Popular';
-          break;
-        default:
-          break;
-      }
-      String filter = prefs.get('FeedFilterSelection');
-      switch (filter) {
-        case 'All':
-          filterType = FilterType.all;
-          _currentSelection = 2;
-          break;
-        case 'Flutter':
-          filterType = FilterType.flutter;
-          _currentSelection = 0;
-          break;
-        case 'Web':
-          filterType = FilterType.web;
-          _currentSelection = 1;
-          break;
-        default:
-          break;
-      }
-    });
-  }
+  ScrollController _scrollController;
 
   @override
   void initState() {
-    _scrollController = ScrollController();
-    loadPreferences();
     super.initState();
+    homeBloc = BlocProvider.of<HomeBloc>(context);
```

Comment:
nit but you could rewrite this like:
```dart
homeBloc = BlocProvider.of<HomeBloc>(context)..dispatch(GetPageOfPackagesEvent());
```

Reply:
Had to remove that anyway, since you have a null context in initState anyway.